partiql-lang/src/test/kotlin/org/partiql/lang/planner/transforms/PartiQLSchemaInferencerTests.kt

```diff
@@ -3209,14 +3210,14 @@ class PartiQLSchemaInferencerTests {
                 catalog = CATALOG_DB,
                 catalogPath = DB_SCHEMA_MARKETS,
                 query = "(order_info.customer_id = 1) AND (order_info.marketplace_id = 2)",
-                expected = TYPE_BOOL
+                expected = StaticType.unionOf(BOOL, NULL)
```

Comment:
Because our function header says the Logical functions AND/OR are `nullable`, the inferencer returns `unionOf(BOOL, NULL)` even for input like `BOOL AND BOOL` OR `NULL AND NULL`. 

The workaround at this stage would be to have a look up table in the inferencer.

Reply:
We could also model special operator functions differently in the unresolved plan since that is internalized. Food for thought, no changes requested at the moment.